src/Api/Controllers/OrganizationsController.cs

```diff
@@ -555,5 +556,62 @@ public async Task PutTaxInfo(string id, [FromBody]OrganizationTaxInfoUpdateReque
             };
             await _paymentService.SaveTaxInfoAsync(organization, taxInfo);
         }
+        
+        [HttpGet("{id}/keys")]
+        public async Task<OrganizationKeysResponseModel> GetKeys(string id)
+        {
+            var user = await _userService.GetUserByPrincipalAsync(User);
+            if (user == null)
+            {
+                throw new UnauthorizedAccessException();
+            }
+            
+            // If the keys aren't populated, error out
+            var org = await _organizationRepository.GetByIdAsync(new Guid(id));
+            if (org == null || org.PublicKey == null || org.PrivateKey == null)
+            {
+                throw new BadRequestException("Organization Keys are not available");
+            }
```

Comment:
Should we just return nulls to the client rather than throwing server errors?

Reply:
I'm okay either way. Just making sure the organization is valid via null check /should/ be enough. Thoughts?

Reply:
Discussed. Will update.

Reply:
Updated.